src/reducers-actions/trades.ts

```diff
@@ -280,3 +238,50 @@ function loadInitialState(): TradesState {
 }
 
 export const initialState = loadInitialState()
+
+// ******** REDUCER
+
+export const reducer = (state: TradesState, action: ReducerActionType): TradesState => {
+  switch (action.type) {
+    case 'APPEND_TRADES': {
+      const { trades: newTrades, reverts, lastCheckedBlock, networkId, userAddress } = action.payload
+
+      const accountKey = buildAccountKey({ networkId, userAddress })
```

Comment:
In pedningOrders we have instead [a nested object](https://github.com/gnosis/dex-react/blob/67f578ce5f0187641297b158e86cca1e2010779d/src/reducers-actions/pendingOrders.ts#L53) `{[networkId]: {[userAddress]:data}}`
But that doesn't matter.
If anything, `accountKey = networkId + '|' + userAddress` allows for a flatter state.
:+1:

Reply:
I was inspired by your key generation for token lists https://github.com/gnosis/dex-react/blob/develop/src/api/tokenList/TokenListApi.ts#L99